concordium-contracts-common/src/types.rs

```diff
@@ -692,6 +750,27 @@ pub enum Address {
     Contract(ContractAddress),
 }
 
+#[cfg(feature = "concordium-quickcheck")]
+impl quickcheck::Arbitrary for Address {
+    fn arbitrary(g: &mut Gen) -> Address {
+        //Randomly pick account or contract address.
+        if quickcheck::Arbitrary::arbitrary(g) {
+            Address::Account(quickcheck::Arbitrary::arbitrary(g))
+        } else {
+            Address::Contract(quickcheck::Arbitrary::arbitrary(g))
+        }
+    }
+
+    fn shrink(&self) -> Box<dyn Iterator<Item = Self>> {
+        match self {
+            Address::Account(a) => Box::new(quickcheck::Arbitrary::shrink(a).map(Address::Account)),
+            Address::Contract(a) => {
+                Box::new(quickcheck::Arbitrary::shrink(a).map(Address::Contract))
+            }
+        }
```

Comment:
Please explain that we make a choice here that when shrinking an account address we only produce account addresses, and vice-versa.

This is a reasonable choice, but not the only reasonable choice :)

Reply:
Done.